tests/test_splitter.py

```diff
@@ -0,0 +1,388 @@
+        dist_val = stats['val']['distribution']
+        for pid in train_ids:
+            total = counts[int(pid)]
+            self.assertEqual(int(total * 0.5 / 0.7), dist_train[pid][0])
+            self.assertEqual(int(total * 0.2 / 0.7), dist_val[pid][0])
+
+        test_ids = stats['test']['values present']
+        self.assertEqual(3, len(test_ids))
+        self.assertEqual(test_ids, stats['gallery']['values present'])
+        self.assertEqual(test_ids, stats['query']['values present'])
+
+        dist_test = stats['test']['distribution']
+        dist_gallery = stats['gallery']['distribution']
+        dist_query = stats['query']['distribution']
+        for pid in test_ids:
+            total = counts[int(pid)]
+            self.assertEqual(total, dist_test[pid][0])
+            self.assertEqual(int(total * 0.3 / 0.7), dist_gallery[pid][0])
+            self.assertEqual(int(total * 0.4 / 0.7), dist_query[pid][0])
+
+    def _generate_detection_dataset(self, **kwargs):
+        append_bbox = kwargs.get("append_bbox")
+        with_attr = kwargs.get("with_attr", False)
+        nimages = kwargs.get("nimages", 10)
+
+        def _get_subset():
+            return np.random.choice(['', 'a', 'b'], p = [0.5, 0.3, 0.2])
+
+        label_cat = LabelCategories()
+        for i in range(6):
+            label = "label{}".format(i+1)
+            if with_attr is True:
+                attributes = {"attr0", "attr{}".format(i+1)}
+            else:
+                attributes = {}
+            label_cat.add(label, attributes = attributes)
+        categories = { AnnotationType.label: label_cat }
+
+        iterable = []
+        attr_val = 0
+        totals = np.zeros(3)
+        for img_id in range(nimages):
+            cnts = np.random.randint(0, 5, 3)
+            totals += cnts
+            annotations = []
+            for label_id, count in enumerate(cnts):
+                attributes = {}
+                if with_attr:
+                    attr_val += 1
+                    attributes["attr0"] = attr_val % 3
+                    attributes["attr{}".format(label_id+1)] = attr_val % 2
+                for ann_id in range(count):
+                    append_bbox(annotations, label_id=label_id, \
+                        ann_id=ann_id, attributes=attributes)
+            item = DatasetItem(
+                id=str(img_id),
+                annotations=annotations,
+                subset=_get_subset(),
+                attributes={"id":img_id}
+            )
+            iterable.append(item)
+
+        dataset = Dataset.from_iterable(iterable, categories)
+        return dataset, totals
+
+    def test_split_for_detection(self):
+        def append_bbox_coco(annotations, **kwargs):
+            annotations.append(Bbox(1,1,2,2, label=kwargs['label_id'],
+                                id=kwargs['ann_id'],
+                                attributes=kwargs['attributes'],
+                                group=kwargs['label_id']))
```

Comment:
In `COCO` (and in general) `group` is expected to match `id`. A group is supposed to represent a single object, having multiple annotations (polygons, boxes, (key-)points, masks etc.).

Reply:
Oh.. I missed it. You're right. group should be same to ann_id.  Thanks.